nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,189 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.webhook;
+  defaultUser = "webhook";
+
+  hooks-file = pkgs.writeText "webhook-config" (
+    builtins.toJSON (
+      builtins.attrValues cfg.hooks
+    )
+  );
+
+  # Todo: more expressive type?
+  typeTriggerRule = types.attrs;
+  
+in {
+  options = {
+    services.webhook = {
+      enable = mkEnableOption "Webhook is a lightweight configurable tool written in Go, that allows you to easily create HTTP endpoints (hooks) on your server, which you can use to execute configured commands.";
+      
+      package = mkOption {
+        type = types.package;
+        default = pkgs.webhook;
+        defaultText = "pkgs.webhook";
+        example = literalExample "pkgs.webhook";
+        description = ''
+          Webhook package to use.
+        '';
+      };
+      user = mkOption {
+        type = types.str;
+        default = defaultUser;
+        description = ''
+          Webhook will be run under this user (user will be created if it doesn't exist. This can be your user name).
+        '';
+      };
+      group = mkOption {
+        type = types.str;
+        default = defaultUser;
+        description = ''
+          Webhook will be run under this group (group will be created if it doesn't exist. This can be your group name).
+        '';
+      };
+      hotreload = mkOption {
```

Comment:
This will not happen because the configuration file only resides in the nix store.
You could put the file in `/etc` and make this module write it too, perhaps unless `hooks == []` to allow a user-maintained mutable file.